docs/architecture.md

```diff
@@ -0,0 +1,94 @@
+
+Three types of application containers are run within your environment:
+
+* **Web containers** handle any incoming traffic from web requests, and serve the website up to your users. These automatically scale to meet traffic demands, with limitations applied based on your plan.
+
+* The **sandbox container** is a copy of the web container with some access controls relaxed, and is used for [maintenance tasks via CLI access](./dashboard/cli.md)
+
+* **Cavalcade containers** handle [running scheduled and background tasks](./scheduled-tasks.md). These automatically scale based on the number of pending tasks, allowing huge numbers of tasks to be run simultaneously.
+
+Containers are treated as "throwaway" instances, which contain a copy of your codebase and Altis modules. No data is stored in application containers, allowing problematic containers to be removed and replcaed at any time. Containers can also be added or removed dynamically, handling load during high traffic events (known as "horizontal scaling").
+
+Traffic is routed to containers through a load balancer. This load balancer ensures traffic is distributed across containers evenly, and ensures that containers being added or removed are seamlessly added to the pool. The load balancer also acts as the firewall and ingress point to your environment, with strict security controls at the network and application layers.
+
+Web containers automatically collect application performance monitoring (APM) data, which is available through the Altis Dashboard.
+
+
+## Database servers
+
+Each environment contains at least one MySQL database server, which is the "writer" instance. Production environments on HA plans also include a minimum of one replica instance, which is used as a "reader".
+
+Database servers will automatically scale based on load, adding and removing additional reader instances as needed (known as "horizontally scaling"). Reader instances are automatically distributed across availability zones, and can be promoted to writers for instant disaster recovery.
+
+Database servers can also be allocated more resources (known as "vertical scaling"). The Altis team manages your database instance type based on your typical load patterns and your environment plan. On higher tiers, your server infrastructure can also be allocated a database proxy, which acts as a load balancer for database connections, improving performance.
+
+
+## Object caching servers
+
+Each environment contains at least one Redis server, used for persistent object caching. This server acts as a key-value cache store, offloading expensive queries from the database.
+
+Object cache servers automatically scale based on load, just like the database servers. Altis automatically handles horizontally scaling these servers based on load, and the Altis team manages your instance type.
```

Comment:
As noted, this is not the case, we don't horizontally scale, we have a fail-over.

Also, I thikn we should be mention it's the WP Object Cache specifically that's using the "object caching servers"